packages/webdriverio/src/commands/page/setViewport.ts

```diff
@@ -0,0 +1,61 @@
+/// <reference path="../../types.ts" />
+import type { SetViewportOptions } from '../browser/setViewport.js'
+
+const minWindowSize = 0
+const maxWindowSize = Number.MAX_SAFE_INTEGER
+
+/**
+ * Resizes the browser viewport within the browser. As oppose to `setWindowSize`,
+ * this command changes the viewport size, not the window size.
+ *
+ * <example>
+ * :setWindowSize.js
+    it('should set viewport to emulate iPhone 15', async () => {
+        const page = await browser.url('https://webdriver.io')
+        await page.setViewport({
+            width: 393,
+            height: 659,
+            deviceScaleFactor: 3
+        });
+    });
+ * </example>
+ *
+ * @alias page.setWindowSize
+ * @param {SetViewportOptions} options                  command arguments
+ * @param {number}             options.width            viewport width in pixels
+ * @param {number}             options.height           viewport height in pixels
+ * @param {number}             options.devicePixelRatio pixel ratio of the viewport
+ * @return {Promise<void>}
+ * @type window
+ */
+export async function setViewport(
+    this: WebdriverIO.Page,
+    options: SetViewportOptions
+) {
+    /**
+     * type check
+     */
+    if (typeof options.width !== 'number' || typeof options.height !== 'number') {
+        throw new Error('setViewport expects width and height of type number')
+    }
+
+    /**
+     * value check
+     */
+    if (options.width < minWindowSize || options.width > maxWindowSize || options.height < minWindowSize || options.height > maxWindowSize) {
+        throw new Error('setViewport expects width and height to be a number in the 0 to 2^31 − 1 range')
+    }
+
+    if (options.devicePixelRatio && (typeof options.devicePixelRatio !== 'number' || options.devicePixelRatio < 0)) {
```

Comment:
let's move the type checking to the top where we already do the type checking for the window size